### PR TITLE
prov/efa: fix a bug in setting info->dest_addr and add a test

### PR DESCRIPTION
**Pull request by wzamazon**
This patch fixed a bug in setting `info->dest_addr` and added a test for the case.

### CODE REVIEW

File: fabtests/pytest/efa/efa_common.py [hunk 1]
```diff
@@ -51,3 +49,20 @@ def has_gdrcopy(hostname):
     command = "ssh {} /usr/sbin/lsmod | grep gdrdrv".format(hostname)
     process = subprocess.run(command, shell=True, check=False, stdout=subprocess.PIPE)
     return process.returncode == 0
+
+def efa_retrieve_gid(hostname):
```
Comment (tschuett) [inline fabtests/pytest/efa/efa_common.py:53]: Does it  retrieve the gid or checks for gdrcopy?

File: fabtests/pytest/efa/efa_common.py [hunk 2]
```diff
@@ -51,3 +49,20 @@ def has_gdrcopy(hostname):
...
+    """
+    determine whether a host has gdrcopy installed
+    hostname: a host
+    return: a string if the host has efa device,
+            None otherwise
+    """
+    command = "ssh {} /usr/sbin/lsmod | grep gdrdrv".format(hostname)
+    command = "ssh {} ibv_devinfo  -v | grep GID | awk '{{print $NF}}' | head -n 1".format(hostname)
```
Comment (tschuett) [inline fabtests/pytest/efa/efa_common.py:61]: You set the command variable twice?

File: fabtests/pytest/efa/efa_common.py (hunk 1 above)
Comment (tschuett) [reply fabtests/pytest/efa/efa_common.py:53]: The docstring is confusing.

**Comment by wzamazon** [comment]
bot:aws:retest

File: fabtests/pytest/efa/efa_common.py [hunk 3]
```diff
@@ -51,3 +49,19 @@ def has_gdrcopy(hostname):
     command = "ssh {} /usr/sbin/lsmod | grep gdrdrv".format(hostname)
     process = subprocess.run(command, shell=True, check=False, stdout=subprocess.PIPE)
     return process.returncode == 0
+
+def efa_retrieve_gid(hostname):
+    """
+    determine whether a host has gdrcopy installed
```
Comment (tschuett) [inline fabtests/pytest/efa/efa_common.py:55]: The docstring talks about `gdrcopy` but the function implies that you want to retrieve the gid.

File: fabtests/pytest/efa/test_efa_info.py [hunk 4]
```diff
@@ -1,7 +1,17 @@
 import pytest
+import copy
```
Comment (wenduwan) [inline fabtests/pytest/efa/test_efa_info.py:2]: Unused import

Comment (wzamazon) [reply fabtests/pytest/efa/test_efa_info.py]: Thank you! fixed.

File: fabtests/pytest/efa/efa_common.py [hunk 5]
```diff
@@ -51,3 +49,19 @@ def has_gdrcopy(hostname):
     command = "ssh {} /usr/sbin/lsmod | grep gdrdrv".format(hostname)
     process = subprocess.run(command, shell=True, check=False, stdout=subprocess.PIPE)
     return process.returncode == 0
+
+def efa_retrieve_gid(hostname):
```
Comment (wckzhang) [inline fabtests/pytest/efa/efa_common.py:53]: How does this work with multiple efa devices?

Comment (wzamazon) [reply fabtests/pytest/efa/efa_common.py:53]: This function always return the gid of the 1st EFA device.

**Review by wenduwan — APPROVED** [review]

File: fabtests/pytest/efa/test_efa_info.py [hunk 6]
```diff
@@ -1,7 +1,16 @@
...
+from efa_common import efa_retrieve_gid
 
 @pytest.mark.unit
 def test_efa_info(cmdline_args):
-    from common import UnitTest
     test = UnitTest(cmdline_args, "fi_efa_info_test")
     test.run()
+
+@pytest.mark.unit
+def test_comm_getinfo(cmdline_args):
```
Comment (shijin-aws) [inline fabtests/pytest/efa/test_efa_info.py:11]: So before this bug fix, this test is actually not run for efa? and the bug is not caught?

**Review by shijin-aws — APPROVED** [review]

File: fabtests/pytest/efa/test_efa_info.py (hunk 6 above)
Comment (wzamazon) [reply fabtests/pytest/efa/test_efa_info.py:11]: The test was run, but the dest address used was ip address. efa does not recorganize that address, therefore return FI_ENODATA (61). `fi_getinfo_test`t consider FI_ENODATA as a valid return.

**Comment by wzamazon** [comment]
AWS CI was sockets provider failed due to hardware issue, not related to this PR.

continuous-integration/appveyor/pr  failure was also sockets provider on windows. Not related to this PR either.

**Pull request merged**